SYCL/SubGroup/sub_groups_sycl2020.cpp

```diff
@@ -0,0 +1,52 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+#include <sycl/sycl.hpp>
+
+class TestKernel;
+
+int main() {
+  sycl::queue Q;
+  sycl::buffer<int, 3> Buf{sycl::range{3, 32, 32}};
+
+  sycl::kernel_bundle KernelBundle =
+      sycl::get_kernel_bundle<sycl::bundle_state::executable>(Q.get_context());
+
+  Q.submit([&](sycl::handler &CGH) {
+    CGH.use_kernel_bundle(KernelBundle);
+    sycl::accessor Acc{Buf, CGH, sycl::write_only};
+    CGH.parallel_for<TestKernel>(
+        sycl::nd_range<1>(sycl::range{32}, sycl::range{32}),
+        [=](sycl::nd_item<1> item) {
+          auto SG = item.get_sub_group();
+          Acc[0][SG.get_group_linear_id()][SG.get_local_linear_id()] =
+              SG.leader();
+          Acc[1][SG.get_group_linear_id()][SG.get_local_linear_id()] =
+              SG.get_group_linear_range();
+          Acc[2][SG.get_group_linear_id()][SG.get_local_linear_id()] =
+              SG.get_local_linear_range();
+        });
+  });
+
+  sycl::host_accessor Acc{Buf, sycl::read_only};
+
+  sycl::kernel_id TestKernelID = sycl::get_kernel_id<TestKernel>();
+  sycl::kernel Kernel = KernelBundle.get_kernel(TestKernelID);
+
+  const size_t MaxNumSubgroups =
+      Kernel.get_info<sycl::info::kernel_device_specific::max_num_sub_groups>(
+          Q.get_device());
+  const size_t SubgroupSize = 32 / MaxNumSubgroups;
```

Comment:
Any reason not to use `info::kernel_device_specific::max_sub_group_size` instead?

Reply:
That's actually a nice piece of advice. `kernel_device_specific::max_num_sub_groups` returns maximum number of subroups, not the one that was actually run. I updated implementation to use the correct values.